doc/guide.md

```diff
@@ -0,0 +1,241 @@
+Writing a libtock-c System Call Driver
+======================================
+
+This guide covers how to implement support in `libtock` for a new system call
+driver.
+
+
+## Setup
+
+
+- `[name]`: String which identifies the system call driver.
+- `[category]`: The general type of the system call.
+
+
+## Internal APIs
+
+All direct system calls must be wrapped in simple functions.
+
+| Characteristic   | Value                                           |
+|------------------|-------------------------------------------------|
+| Location         | `libtock/[category]/internal`                   |
+| Source File Name | `libtock/[category]/internal/[name].c`          |
+| Header File Name | `libtock/[category]/internal/[name]_internal.h` |
```

Comment:
What is the reasoning for `[name]_internal.h` vs `[name].h`?

Reply:
I think I was thinking so it wouldn't conflict with the "non internal" header. But the directory structure should take care of that.

Reply:
Definitely syscalls, and its probably worth not having multiple files with the same name, even if we could get away with it.